File: org-frontend/src/api/tree.ts
import { client } from "./client"
import { commonHeaders } from "./utils"

// Types for the tree structure from the backend
export interface TreeNote {
  id: string
  title: string
}

export interface TreeSection {
  id: string
  title: string
  notes: TreeNote[]
}

export interface TreeNotebook {
  id: string
  title: string
  sections: TreeSection[]
  unsectioned: TreeNote[]
}

export interface TreeData {
  notebooks: TreeNotebook[]
  unassigned: TreeNote[]
}

export const treeClient = {
  fetch: async (): Promise<TreeData> => {
    return await client
      .get("tree", {
        headers: commonHeaders(),
        credentials: "include",
      })
      .json<TreeData>()
  },
}

File: org-go/pkg/models/tree.go
package models

import "github.com/google/uuid"

// TreeNote represents a minimal note for the tree view
type TreeNote struct {
	ID    uuid.UUID `json:"id"`
	Title string    `json:"title"`
}

// TreeSection represents a section with its notes for the tree view
type TreeSection struct {
	ID    uuid.UUID  `json:"id"`
	Title string     `json:"title"`
	Notes []TreeNote `json:"notes"`
}

// TreeNotebook represents a notebook with sections and unsectioned notes for the tree view
type TreeNotebook struct {
	ID          uuid.UUID     `json:"id"`
	Title       string        `json:"title"`
	Sections    []TreeSection `json:"sections"`
	Unsectioned []TreeNote    `json:"unsectioned"`
}

// TreeData represents the complete tree structure
type TreeData struct {
	Notebooks  []TreeNotebook `json:"notebooks"`
	Unassigned []TreeNote     `json:"unassigned"`
}
